libsignal-service/src/cipher.rs

```diff
@@ -19,23 +21,31 @@ use prost::Message;
 /// Decrypts incoming messages and encrypts outgoing messages.
 ///
 /// Equivalent of SignalServiceCipher in Java.
-#[derive(Clone, Debug)]
+#[derive(Clone)]
 pub struct ServiceCipher {
+    pub(crate) context: Context,
     pub(crate) store_context: StoreContext,
     pub(crate) local_address: ServiceAddress,
-    pub(crate) context: Context,
+    sealed_session_cipher: SealedSessionCipher,
 }
 
 impl ServiceCipher {
     pub fn from_context(
         context: Context,
-        local_address: ServiceAddress,
         store_context: StoreContext,
+        local_address: ServiceAddress,
+        certificate_validator: CertificateValidator,
```

Comment:
Would it make sense to construct the `CertificateValidator` here, based on `context`?

Reply:
well, the CertificateValidator needs a trusted root certificate which comes from ServiceConfiguration. I'm fine passing this as a parameter instead, as it's already done for AwcPushService.